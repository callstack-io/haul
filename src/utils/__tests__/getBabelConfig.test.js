/**
 * Copyright 2017-present, Callstack.
 * All rights reserved.
 *
 * @flow
 */

const path = require('path');
const getBabelConfig = require('../getBabelConfig');
const traverse = require('traverse');

const orignalEnv = process.env.NODE_ENV;

afterEach(() => {
  process.env.NODE_ENV = orignalEnv;
});

test('extracts config from .babelrc', () => {
  const cwd = path.resolve(__dirname, './fixtures');
  const config = getBabelConfig(cwd);
  expect(config).toMatchObject({
    extends: path.resolve(cwd, '.babelrc'),
  });
});

test('creates new config when no .babelrc found', () => {
  const cwd = path.resolve('mocked/path');
  const config = getBabelConfig(cwd);
  expect(
    traverse(config).map(
      entry =>
        typeof entry === 'string'
          ? entry
              .replace(/\/.*\/src/, '<<REPLACED>>')
              .replace(/\/.*\/node_modules/, '<<NODE_MODULE>>')
          : entry,
    ),
  ).toMatchSnapshot();
});

test('does not include "hot" plugins in production', () => {
  process.env.NODE_ENV = 'production';
  const cwd = path.resolve('mocked/path');
  const config = getBabelConfig(cwd);
  expect(
    traverse(config).map(
      entry =>
        typeof entry === 'string'
          ? entry
              .replace(/\/.*\/src/, '<<REPLACED>>')
              .replace(/\/.*\/node_modules/, '<<NODE_MODULE>>')
          : entry,
    ),
  ).toMatchSnapshot();
});
