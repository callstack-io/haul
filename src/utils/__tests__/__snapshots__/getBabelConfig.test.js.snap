// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates new config when no .babelrc found 1`] = `
Object {
  "plugins": Array [
    "<<REPLACED>>/src/utils/fixRequireIssues.js",
    "<<NODE_MODULE>>/react-hot-loader/babel.js",
    "<<REPLACED>>/src/hot/babelPlugin.js",
  ],
  "presets": Array [
    Array [
      "<<NODE_MODULE>>/metro-react-native-babel-preset/src/index.js",
      Object {
        "disableImportExportTransform": true,
      },
    ],
  ],
}
`;

exports[`does not include "hot" plugins in production 1`] = `
Object {
  "plugins": Array [
    "<<REPLACED>>/src/utils/fixRequireIssues.js",
  ],
  "presets": Array [
    Array [
      "<<NODE_MODULE>>/metro-react-native-babel-preset/src/index.js",
      Object {
        "disableImportExportTransform": true,
      },
    ],
  ],
}
`;

exports[`does not include "hot" plugins when HMR is disabled 1`] = `
Object {
  "plugins": Array [
    "<<REPLACED>>/src/utils/fixRequireIssues.js",
  ],
  "presets": Array [
    Array [
      "<<NODE_MODULE>>/metro-react-native-babel-preset/src/index.js",
      Object {
        "disableImportExportTransform": true,
      },
    ],
  ],
}
`;
